src/ui/public/courier/saved_object/saved_object.js

```diff
@@ -299,15 +313,29 @@ export function SavedObjectProvider(Promise, Private, Notifier, confirmModalProm
       if (this.searchSource) {
         const searchSourceFields = _.omit(this.searchSource.getFields(), ['sort', 'size']);
         if (searchSourceFields.index) {
-          const indexId = searchSourceFields.index;
+          const { id: indexId } = searchSourceFields.index;
```

Comment:
`searchSourceFields.index` is always an instance of `IndexPattern`. The `toJSON` function in that class always replaced the object with the `id` value after stringification which made things work normally. This has been caught while writing tests and I'm changing it to set only the `id` in the references.